aten/src/ATen/native/Convolution.cpp

```diff
@@ -985,6 +982,25 @@ static inline std::vector<int64_t> calc_output_size(
   return output_size;
 }
 
+static inline at::MemoryFormat determine_backend_memory_format(
+    const Tensor& input,
+    const Tensor& weight) {
+  at::MemoryFormat backend_memory_format = at::MemoryFormat::Contiguous;
+  auto k = weight.ndimension();
+#if !defined(C10_MOBILE)
+  // cudnn and miopen are guaranteed not to be on mobile, and T102591915
+  // suggests that maybe the cudnn condition sometimes segfaults (though
+  // I can't imagine how)
+  if (detail::getCUDAHooks().compiledWithCuDNN()) {
+    backend_memory_format = cudnn_conv_suggest_memory_format(input, weight);
+  }
+  if (detail::getCUDAHooks().compiledWithMIOpen() && miopen_conv_use_channels_last(input, weight)) {
+    backend_memory_format = (k == 5) ? at::MemoryFormat::Contiguous /*at::MemoryFormat::ChannelsLast3d*/ : at::MemoryFormat::ChannelsLast;
```

Comment:
Is this assuming that k == 4 or 5 here?

Reply:
Yep looks like it - it's technically valid at this point since the k == 3 case is explicitly expanded to 4D before this is called.

Reply:
can we add an internal assert on k == 4 or 5?